pennylane_qiskit/qiskit_device2.py

```diff
@@ -372,8 +369,37 @@ def preprocess(
 
         return transform_program, config
 
-    @staticmethod
-    def get_transpile_args(kwargs):
+    def _process_kwargs(self):
+        """Combine the settings defined in options and the settings passed as kwargs, with
+        the definition in options taking precedence if there is conflicting information.
+        Arguments related to transpilation are separated and saved in `dev._transpile_args`."""
+
+        if "noise_model" in self._kwargs:
+            noise_model = self._kwargs.pop("noise_model")
+            self.backend.set_options(noise_model=noise_model)
+
+        if "options" in self._kwargs:
+            for key, val in self._kwargs.pop("options").items():
+                if key in self._kwargs:
+                    warnings.warn(
+                        "An overlap between what was passed in via options and what was passed in via kwargs was found."
+                        f"The value set in options {key}={val} will be used."
+                    )
+                self._kwargs[key] = val
+
+        shots = self._kwargs.pop("shots")
+
+        if "default_shots" in self._kwargs:
+            warnings.warn(
+                f"default_shots was found in the keyword arguments, but it is not supported by {self.name}"
+                "Please use the `shots` keyword argument instead. The number of shots "
+                f"{shots} will be used instead."
+            )
+        self._kwargs["default_shots"] = shots
+
+        self._transpile_args = self.get_transpile_args()
+
+    def get_transpile_args(self):
```

Comment:
Same suggestion for this - 
```suggestion
    def get_transpile_args(self, kwargs): --> _transpile_args
```
That it processes the `kwargs` given to it and not the ones contained in the object directly.